ui/src/components/core/combobox/combobox.tsx

```diff
@@ -0,0 +1,126 @@
+import { Fragment, useRef, useState } from 'react';
+import { Combobox as ComboboxLib, Transition } from '@headlessui/react';
+import { Icon, IconName } from '@/components/core/icon';
+import { Flex } from '@/components/layout';
+
+export type ComboboxItem = {
+  value: string;
+  label: string;
+  icon?: IconName;
+};
+
+export type ComboboxProps = {
+  items: ComboboxItem[];
+  selectedValue: ComboboxItem | undefined;
+  onChange(option: ComboboxItem): void;
+};
+
+export const Combobox: React.FC<ComboboxProps> = ({
+  items,
+  selectedValue = { value: '', label: '' },
+  onChange,
+}) => {
+  const [query, setQuery] = useState('');
+
+  const buttonRef = useRef<HTMLButtonElement>(null);
+
+  const filteredItems =
+    query === ''
+      ? items
+      : items.filter((person) =>
+          person.label
+            .toLowerCase()
+            .replace(/\s+/g, '')
+            .includes(query.toLowerCase().replace(/\s+/g, ''))
+        );
+
+  return (
+    <ComboboxLib
+      value={selectedValue}
+      by="value"
+      onChange={(option: ComboboxItem) => onChange(option)}
+    >
+      {({ open }) => (
+        <>
+          <div className="w-full cursor-default overflow-hidden bg-transparent text-left focus:outline-none sm:text-sm">
+            <div className="relative">
+              <Icon
+                name="search"
+                size="sm"
+                css={{
+                  position: 'absolute',
+                  left: '$3',
+                  top: '0.9375rem',
+                  color: 'slate8',
+                }}
+              />
+              <ComboboxLib.Input
+                placeholder="Search"
+                className={`w-full border-solid border border-slate7  py-3 pl-8 pr-10 text-sm bg-transparent leading-5 text-slate11 outline-none ${
+                  open
+                    ? 'border-b-0 rounded-t-xl bg-black border-slate6'
+                    : 'rounded-xl'
+                }`}
+                displayValue={(selectedValue: ComboboxItem) =>
+                  selectedValue.label
+                }
+                onChange={(event) => setQuery(event.target.value)}
+                onClick={() => buttonRef.current?.click()}
+              />

```

Comment:
It would be cool to remove the handlers from inside the render to keep the code cleaner.